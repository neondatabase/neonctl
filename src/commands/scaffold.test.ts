import { describe } from 'vitest';

import { test } from '../test_utils/fixtures';

describe('scaffold', () => {
  test('list', async ({ testCliCommand }) => {
    await testCliCommand(['scaffold', 'list']);
  });

  test('start validate template is set', async ({ testCliCommand }) => {
    await testCliCommand(['scaffold', 'start'], {
      code: 1,
    });
  });

  test('start with project id', async ({ testCliCommand }) => {
    await testCliCommand([
      'scaffold',
      'start',
      'test-template',
      '--project-id',
      'test',
    ]);
  });

  test('start without project id', async ({ testCliCommand }) => {
    await testCliCommand(['scaffold', 'start', 'test-template']);
  });
});
