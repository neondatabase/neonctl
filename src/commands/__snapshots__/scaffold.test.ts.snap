// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`scaffold > list 1`] = `
"- name: astro
- name: laravel-11
"
`;

exports[`scaffold > start validate template is set 1`] = `""`;

exports[`scaffold > start with project id 1`] = `""`;

exports[`scaffold > start without project id 1`] = `""`;
